src/forms/field_mapper.ts

```diff
@@ -0,0 +1,219 @@
+}
+
+export class FieldMapper implements Imapper {
+    private values: AppCallValues
+    private postMessage: string
+    constructor(values: AppCallValues, postMessage: string) {
+        this.values = values;
+        this.postMessage = postMessage;
+    }
+    mapFieldsToAppFields(fields: UserField[]): AppField[] {
+        const appFields: AppField[] = [];
+        fields.forEach((field) => {
+            switch (field.type) {
+            // case 'assignee':
+            // appFields.push(field.title);
+            // return;
+            // case zdTypeNumeric:
+            //     appFields.push(mapToNumeric(field, true));
+            //     return;
+            case zdTypes.zdTypeSubject: // system type
+            case zdTypes.zdTypeText: // custom type
+            case zdTypes.zdTypeMultiLine: // custom type
+                appFields.push(this.mapToText(field));
+                return;
+            case zdTypes.zdTypeCheckbox: // will always be custom??
+                appFields.push(this.mapToBool(field));
+                return;
+            case zdTypes.zdTypeDescription: // will be filled by post message
+                return;
+            case zdTypes.zdTypeTicketType:
+            case zdTypes.zdTypePriority:
+            case zdTypes.zdTypeTagger: // system or custom type
+                appFields.push(this.mapToStaticSelect(field));
+            default:
+                // console.log('field not mapped to app field. field = ', field);
+                break;
+            }
+        });
+
+        // append optional message and post message to end of form fields
+        const postFields = this.getPostFields(this.postMessage);
+        appFields.push(...postFields);
+
+        return appFields;
+    }
+
+    // mapFormsToSelectField maps zendesk forms to a Mattermost select field
+    mapFormsToSelectField(forms: any): AppField {
+        const name = fieldNames.formsSelectName;
+        const selectField: AppField = {
+            name,
+            label: 'Form',
+            type: AppFieldTypes.STATIC_SELECT,
+            options: makeFormOptions(forms),
+            value: this.getSavedValue(name),
+            is_required: true,
+            refresh: true,
+        };
+        return selectField;
+    }
+
+    // mapToBool maps Zendesk checkbox field types to the Mattermost bool field
+    private mapToBool(field: UserField): AppField {
+        const name = this.getMappedCustomName(field);
+        const boolField: AppField = {
+            name,
+            label: field.title,
+            type: AppFieldTypes.BOOL,
+            is_required: field.required_in_portal,
+            value: this.getSavedValue(name),
+        };
+        return boolField;
+    }
+
+    // mapToStaticSelect maps Zendesk field types to the Mattermost static select field
+    private mapToStaticSelect(field: UserField): AppField {
+        let options: number[];
+        let name: string;
+
+        switch (field.type) {
```

Comment:
Remove switch.  Add logic to isSystemField method to determine which what value to use for options variable

Reply:
done.